src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -96,6 +96,16 @@ public override BoundNode VisitBaseReference(BoundBaseReference node)
             return base.VisitBaseReference(node);
         }
 
+        public override BoundNode VisitSwitchExpression(BoundSwitchExpression node)
+        {
+            if (_inExpressionLambda)
+            {
+                Error(ErrorCode.ERR_ExpressionTreeContainsSwitchExpression, node);
```

Comment:
>ERR_ExpressionTreeContainsSwitchExpression [](start = 32, length = 42)

Is there a test for this code path?

Reply:
A test will be added in a forthcoming iteration.

---
In reply to: [177542903](https://github.com/dotnet/roslyn/pull/25697#discussion_r177542903) [](ancestors = 177542903)